packages/core/js-client/src/compilerSupport/callFunction.ts

```diff
@@ -44,9 +45,8 @@ export const callAquaFunction: CallAquaFunctionType = ({ def, script, config, pe
     log.trace('calling aqua function %j', { def, script, config, args });
     const argumentTypes = getArgumentTypes(def);
 
-    const promise = new Promise((resolve, reject) => {
-        const particle = peer.internals.createNewParticle(script, config?.ttl);
 
+    return peer.internals.createNewParticle(script, config?.ttl).then((particle: IParticle) => new Promise((resolve, reject) => {
```

Comment:
oh. didn't notice. why you decided not to just use async/await here

Reply:
thanks, changed it